packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1012,25 +1033,53 @@ class TextureAndroidViewController extends AndroidViewController {
   @override
   int? get textureId => _textureId;
 
-  late Size _size;
+  late Size _initialSize;
+
+  /// The current offset of the platform view.
+  Offset _off = Offset.zero;
 
   @override
-  Future<void> setSize(Size size) async {
+  Future<Size> setSize(Size size) async {
```

Comment:
This is a breaking change. Can we just add a new method to get the updated size?

Reply:
This is necessary. When someone overrides `setSize`, they do it for testing purposes. 
e.g. to provide a fake implementation of this controller. If there's a new method that makes calls over the channel, then failures are expected as a result.

Reply:
Why would it be necessary, can we have an ivar to save it after calling `void setSize`, and a getter to get the value of that ivar?

It is also a little confusing to get a new Size object when you setSize.

Reply:
Can we query all flutter projects to make sure there's no subclasses of AndroidViewController that overrides `setSize`?

Reply:
I didn't see any subclasses other than `FakeAndroidViewController` already defined within the framework (google3 and public repos). In general, this controller is an internal implementation of the framework; and it's very tightly coupled to the platform implementation.

I'm also not aware of any use case (other than maybe a fake?) that requires overriding or extending functionality.

Reply:
At a high level, this API seems like it should have been private. I don't know why it was made public.